port/osx/omrosbacktrace_impl.c

```diff
@@ -0,0 +1,413 @@
+				/* increment ret to the number of filled addresses */
+			}
+
+			threadInfo->error = FAULT_DURING_BACKTRACE;
+		}
+	} else {
+		ret = unw_backtrace(address_array, capacity);
+	}
+
+	return ret;
+}
+
+/**
+ * This function constructs a backtrace from a CPU context. Generally there are only one or two
+ * values in the context that are actually used to construct the stack but these vary by platform
+ * so arn't detailed here. If no heap is specified then this function will use malloc to allocate
+ * the memory necessary for the stack frame structures which must be freed by the caller.
+ *
+ * @param portLibrary a pointer to an initialized port library
+ * @param threadInfo the thread structure we want  to attach the backtrace to. Must not be NULL.
+ * @param heap a heap from which to allocate any necessary memory. If NULL malloc is used instead.
+ * @param signalInfo a platform signal context. If not null the context held in threadInfo is replaced
+ *  	  with the signal context before the backtrace is generated.
+ *
+ * @return the number of frames in the backtrace.
+ */
+uintptr_t
+omrintrospect_backtrace_thread_raw(OMRPortLibrary *portLibrary, J9PlatformThread *threadInfo, J9Heap *heap, void *signalInfo)
+{
+	void *addresses[50];
+	J9PlatformStackFrame **nextFrame = NULL;
+	J9PlatformStackFrame *junkFrames = NULL;
+	J9PlatformStackFrame *prevFrame = NULL;
+	OMRUnixSignalInfo *sigInfo = (OMRUnixSignalInfo *)signalInfo;
+	uintptr_t i = 0;
+	uintptr_t discard = 0;
+	uintptr_t ret = 0;
+	const char *regName = NULL;
+	void **faultingAddress = NULL;
+
+	if (NULL == threadInfo) {
+		return 0;
+	}
+
+	/* if we've been passed a port library wrapped signal, then extract info from there */
+	if (NULL != sigInfo) {
+		/* get the faulting address so we can discard frames that are part of the signal handling */
+		infoForControl(portLibrary, sigInfo, OMRPORT_SIG_CONTROL_PC, &regName, (void **)&faultingAddress);
+	}
+
+	ret = backtrace_sigprotect(portLibrary, threadInfo, addresses, sizeof(addresses) / sizeof(addresses[0]));
+
+	nextFrame = &threadInfo->callstack;
+	for (i = 0; i < ret; i++) {
+		if (NULL != heap) {
+			*nextFrame = portLibrary->heap_allocate(portLibrary, heap, sizeof(J9PlatformStackFrame));
+		} else {
+			*nextFrame = portLibrary->mem_allocate_memory(portLibrary, sizeof(J9PlatformStackFrame), OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+		}
+
+		if (NULL == *nextFrame) {
+			if (0 == threadInfo->error) {
+				threadInfo->error = ALLOCATION_FAILURE;
+			}
+
+			break;
+		}
+
+		(*nextFrame)->parent_frame = NULL;
+		(*nextFrame)->symbol = NULL;
+		(*nextFrame)->instruction_pointer = (uintptr_t)addresses[i];
+		(*nextFrame)->stack_pointer = 0;
+		(*nextFrame)->base_pointer = 0;
+
+		nextFrame = &(*nextFrame)->parent_frame;
+
+		/* Check to see if we should truncate the stack trace to omit handler frames. The faulting address may
+		 * be one less than the address reported by core dump and backtrace on osx. */
```

Comment:
`*/` should be on a new line.
```suggestion
		/* Check to see if we should truncate the stack trace to omit handler frames. The faulting address may
		 * be one less than the address reported by core dump and backtrace on osx.
		 */
```